src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -487,6 +490,58 @@ private void RootGrid_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
 			}
 		}
 
+		private bool lockFlag = false;
+		//private string[] dropableArchiveTypes = { "zip", "rar", "7z", "tar" };
+		private async void HorizontalMultitaskingControlAddButton_Drop(object sender, DragEventArgs e)
+		{
+			if (lockFlag || !FilesystemHelpers.HasDraggedStorageItems(e.DataView))
+				return;
+			lockFlag = true;
+
+			var items = (await FilesystemHelpers.GetDraggedStorageItems(e.DataView))
+				.Where(x => x.ItemType is FilesystemItemType.Directory
+				//|| dropableArchiveTypes.Contains(x.Name.Split('.').Last().ToLower())
+				);
+
+			var deferral = e.GetDeferral();
+			try {
```

Comment:
Similar to the above.
```suggestion
			try
			{
```